### ISSUE TITLE
NewQuoteFromYahoo is broken due to Yahoo Finance API changes

### ISSUE THREAD

**Issue by kfwalther** [markcheno/go-quote#37]
The URL used in this method to grab historical finance data (reproduced below) no longer works due to recent Yahoo Finance API changes over the past couple weeks. It now yields an "unauthorized" error with "User is not logged in". 
```
https://query1.finance.yahoo.com/v7/finance/download/%s?period1=%d&period2=%d&interval=1d&events=history&corsDomain=finance.yahoo.com
```

The work-around is to change the following in the URL:

- change "query1" to "query2"
- change "v7" to "v8"
- change "download" to "chart"

Instead of downloading in CSV format, you now get JSON containing same data. I don't currently have the time to update the NewQuoteFromYahoo method to parse the JSON into a new Quote object.

**Comment by markcheno** [comment]
Thanks for the heads up. Unfortunately these api's are a continuously moving target and very hard to maintain.

**Comment by kfwalther** [comment]
@markcheno No problem, it probably won't be the last time they change the API either :)

I've made a PR for this and it now pulls a JSON response and parses it successfully into the Quote object. It's working without issue for the tickers I tested with: a few dozen equities, some ETFs, and some crypto tickers. 

Feel free to make edits / adjustments as you see fit. 
https://github.com/markcheno/go-quote/pull/38
